Added slug to the Frontmatter Reference in the Starlight Docs (#1263)
withastro/starlight: docs/src/content/docs/reference/frontmatter.md

The page description is used for page metadata and will be picked up by search engines and in social media previews.

### `slug`

**type**: `string`

> Overrides the generated slug for the page.

Reviewer: I think we can mention that this property comes from content collections and its not a Starlight specific, this way we could include a link back to the official docs for more information. Let me know what you think.

```suggestion
Content collections reserved property to override the slug of the page, check [defining custom slugs](https://docs.astro.build/en/guides/content-collections/#defining-custom-slugs) for more details.
```

Reviewer 2: Sounds great to me, thank you!

Author: Sounds great to me, thanks!

> Content collections reserved property to override the slug of the page, check [defining custom slugs](https://docs.astro.build/en/guides/content-collections/#defining-custom-slugs) for more details.

Reviewer 3: Thanks for the suggestion @chrskerr! In general we try not to document too much that’s part of Astro’s default functionality, but I agree in this case it’s helpful. Going to make one slight tweak to the proposed text and then should be good to merge.

```suggestion
Override the slug of the page, see [“Defining custom slugs”](https://docs.astro.build/en/guides/content-collections/#defining-custom-slugs) in the Astro docs for more details.
```
